### ISSUE TITLE
Tall style: field formatting

### ISSUE THREAD

**Issue by iinozemtsev** [dart-lang/dart_style#1604]
When the field declaration is long, so that the name has to be wrapped, it used to be wrapped with an indent, like this:

```
class C {
  final LoremIpsumDolorSitAmetFactory
      _loremIpsumDolorSitAmetFactory;
}
```

Now there's no indent, and it is harder to read, especially when there are many fields like that.

```
class C {
  final LoremIpsumDolorSitAmetFactory
  _loremIpsumDolorSitAmetFactory;
}
```

Also when there's a comment between an annotation and a type, field name is put on a new line:

Before:

```
  @override
  // ignore: hash_and_equals
  final int hashCode;
```

Now:

```
  @override
  // ignore: hash_and_equals
  final int
  hashCode;
```

**Comment by munificent** [comment]
> Now there's no indent, and it is harder to read, especially when there are many fields like that.
> 
> ```dart
> class C {
>   final LoremIpsumDolorSitAmetFactory
>   _loremIpsumDolorSitAmetFactory;
> }
> ```

This is working as intended. I wrote a comment about that here: https://github.com/dart-lang/dart_style/issues/1541#issuecomment-2316408292.

There's really no perfect solution when you have a really long type annotation followed by a variable or function name. Indenting always looked weird. Not indenting looks kind of weird too, but after trying both styles on a large corpus, I think the latter is generally better. It's consistent with things like metadata annotations and doc comments which also go on preceding lines but don't cause the variable or function name to be indented.


> Also when there's a comment between an annotation and a type, field name is put on a new line:
> 
> ```
>   @override
>   // ignore: hash_and_equals
>   final int
>   hashCode;
> ```

This is a bug. I'll take a look.

**Comment by munificent** [comment]
I'm going to re-open this because users keep being surprised by this style, which suggests that maybe we should go back to indenting fields/variables if the type annotation splits.

I'm not sure if this also means we should indent *functions* if the return type splits. I'll do some investigation.

**Comment by mattrberry** [comment]
> It's consistent with things like metadata annotations and doc comments which also go on preceding lines but don't cause the variable or function name to be indented.

One way in which this differs is that annotations and comments are easier to distinguish from the front of the line. If I'm reading a line that starts with `@` or `//`, I know what I'm looking at. When it comes to variable names, the new format requires me to scan to the end of the prior line to tell whether I'm looking at a variable name or a statement.

Just a thought as to why this jumped out to me. I agree, though, that neither solution is perfect and that I could probably get used to either :)

**Comment by jakemac53** [comment]
I also found this formatting surprising - without any indentation it looks like two separate statements/declarations to me.

The indentation is I think important to show this is a continuation of the thing on the previous line.

I agree they both look weird, but the indentation at least shows clearly that something weird is going on, where without the indentation I think it is very confusing.

**Comment by jellynoone** [comment]
How should inline Function and record types behave?
```dart
class MyFactory {
  final CustomPainter Function({
    required Color color,
    required Animation<double> animation,
  })
  painter;
  final ({
    Color color,
    Animation<double> animation,
    Color color2,
    Animation<doulbe> animation2,
  })
  painter2;
}
```
I'd expect this to format as:
```dart
class MyFactory {
  final CustomPainter Function({
    required Color color,
    required Animation<double> animation,
  }) painter;
  final ({
    Color color,
    Animation<double> animation,
    Color color2,
    Animation<doulbe> animation2,
  }) painter2;
}
```

**Comment by rakudrama** [comment]
I just came across this for a field with a long function type:
```dart
class Caller {
  final String Function(String argument1, String argument2, String argument3)
  _function;
  Caller(this._function);
}
```
It looks super weird to me that `_function;` is not indented.

**Comment by munificent** [comment]
Yeah, I think I just got this wrong when defining the new tall style. I'd like to circle back and tweak this, but there have been a *lot* of formatter changes in the past couple of releases, so I'm going to let things settle down for a while before I do more formatter work.
